library/src/main/java/org/apache/apex/malhar/lib/state/managed/MovingBoundaryTimeBucketAssigner.java

```diff
@@ -125,9 +125,11 @@ public long getTimeBucket(long time)
       long move = (diffInBuckets + 1) * bucketSpanMillis;
       start += move;
       end += move;
-      lowestPurgeableTimeBucket += diffInBuckets;
       // trigger purge when lower bound changes
-      triggerPurge = (diffInBuckets > 0);
+      triggerPurge = (diffInBuckets >= 0);
```

Comment:
Shouldn't triggerPurge depend on "move" (which is changing start)? We want to make sure that the purging logic is only called when it is needed, even when it is a noop. Also, can you please improve the documentation of this method while the knowledge is fresh. Lack of documentation is a big problem with the managed state component.